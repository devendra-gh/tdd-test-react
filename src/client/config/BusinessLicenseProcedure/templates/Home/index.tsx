import React from 'react';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import JourneyLandingTemplate from '@tamm/ui-lib-v2-journey-landing-template';
import Sidebar from 'client/templates/Sidebar';
import { steps } from '../../steps';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n } = props;
  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <Sidebar
            currentStep={props.currentStep}
            currentSubStep={props.currentSubStep}
            i18n={props.i18n}
            steps={props.steps}
            showRelatedJourneyCard
            stepsStatus={props.stepsStatus}
          />
        }
      >
        <div className="ui-lib-story-container ui-lib-story-container_bg-light-grey">
          <JourneyLandingTemplate
            description={i18n('businessLicenseProcedure.home.description')}
            process={{
              steps: steps.map(step => ({
                description: i18n(`${step.name}.desc`),
                label: i18n(step.name),
              })),
              title: i18n('label.process'),
            }}
            startLogin={{
              buttonLabel: i18n('button.start'),
              description: i18n(
                'businessLicenseProcedure.home.start.description',
              ),
              title: i18n('businessLicenseProcedure.main.title'),
              onClick: () => props.onStart(props),
            }}
            title={i18n('label.serviceDescription')}
            helpfulBlock={{}}
          />
        </div>
        <div style={{ height: 200 }} />
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
