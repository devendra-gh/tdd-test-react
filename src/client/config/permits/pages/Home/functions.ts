/* eslint-disable complexity */
import { IVariables } from '@tamm/app-composer';
import bpm from 'client/services/bpm';
import scrollToElement from 'client/config/utils/permitsScrollToElement';
import permitConfigs from 'client/config/permits/permitConfigs';
import fetchLicenseDetails from './functions/fetchLicenseDetails';
import validation from './functions/validation';
import fetchPermitFees from './functions/fetchPermitFees';
import { errorMsgMap } from '../../constants';
import { fetchRequirementsInquiry } from '../../services';
import {
  PATH_UNDERTAKING,
  PATH_APPLICATION_WAITING,
} from '../../utils/constants/pageRoutes';

const getPermitCompanyTypes = async (serviceName: string) => {
  try {
    const permitRequirements = await fetchRequirementsInquiry(serviceName);
    let permitCompanyTypes = ['DED', 'ADGE', 'FZ', 'NL', 'FC'];
    // We need to explicitly check for "false" from DED API to reduce companyType list
    // Please dont change this
    if (permitRequirements.data.result.AllowedForContactType === false) {
      permitCompanyTypes = ['DED'];
    }
    return permitCompanyTypes;
  } catch (e) {
    return ['DED', 'ADGE', 'FZ', 'NL', 'FC'];
  }
};

const init = async (props: IVariables) => {
  props.actions.permitSubmitting.update(false);
  props.actions.permitServerError.update('');
  props.actions.companyType.update('DED');

  const { serviceName } = props.match.params;

  const PermitConfigs: { [key: string]: any } = permitConfigs;

  const { serviceType } = props;

  const serviceConfig: any = serviceName
    ? PermitConfigs[serviceName]
    : PermitConfigs[serviceType];

  if ((serviceName || serviceType) && serviceConfig) {
    const { name, category, requiresUndertakingApproval } = serviceConfig;
    const { permitInfo } = props;
    const { undertaking } = permitInfo[name];
    const { isApproved } = undertaking;
    const permitFeesDetails = await fetchPermitFees(name);
    const permitCompanyTypes = await getPermitCompanyTypes(name);
    props.actions.permitCompanyTypes.update(permitCompanyTypes);

    if (name !== serviceType) {
      props.actions.serviceType.update(name);
      props.actions.permitType.update(category);
    }

    if (requiresUndertakingApproval && !isApproved) {
      props.actions.permitInfo.update({
        ...permitInfo,
        [name]: {
          ...permitInfo[name],
          undertaking: {
            ...permitInfo[name].undertaking,
            showError: true,
          },
        },
      });
      props.history.push(`/${name}${PATH_UNDERTAKING}`);
    } else {
      props.actions.urlServiceName.update(true);
      props.actions.serviceType.update(name);
      props.actions.permitType.update(category);

      // const { permitInfo } = props;
      const Info = {
        name: props.user['First Name EN'],
        email: props.user['User Email'],
        phone: props.user.Mobile,
      };

      props.actions.permitInfo.update({
        ...permitInfo,
        [name]: {
          ...permitInfo[name],
          applicantContact: {
            ...Info,
          },
          ...permitFeesDetails,
        },
      });
    }
  } else {
    props.actions.urlServiceName.update(false);
  }
};

const onSubmit = (bpmUrl: string) => async (props: IVariables) => {
  const { serviceType, permitInfo, instanceId, user } = props;

  let permitServerError: string = '';
  props.actions.permitSubmitting.update(true);
  props.actions.permitServerError.update(permitServerError);

  const serviceDetails = permitInfo[serviceType];
  const PermitConfigs: { [key: string]: any } = permitConfigs;
  const permitConfig = PermitConfigs[serviceType];
  const { aduServiceKey, formSubmitDetails } = permitConfig;

  const submitData = formSubmitDetails(serviceDetails, props);
  let { businessKey } = props;

  if (!businessKey || !instanceId) {
    try {
      const data = await bpm.start(bpmUrl, {
        serviceName: aduServiceKey,
        emiratesId: user.IDN,
      });
      if (data.success && data.data && data.data.businessKey && data.data.id) {
        props.actions.instanceId.update(data.data.id);
        props.actions.businessKey.update(data.data.businessKey);
        businessKey = data.data.businessKey;
      } else {
        permitServerError = 'something_went_wrong';
      }
    } catch (e) {
      permitServerError = 'something_went_wrong';
    }
  }

  if (businessKey) {
    try {
      const response = await bpm.message(bpmUrl, {
        businessKey,
        messageName: 'onSubmitPermitDetails',
        variables: submitData,
      });
      if (response.success && response.message === 'Success') {
        props.history.push(`/${serviceType}${PATH_APPLICATION_WAITING}`);
      } else if (response.success && response.message) {
        permitServerError = errorMsgMap[response.message]
          ? errorMsgMap[response.message]
          : 'something_went_wrong';
      } else {
        permitServerError = 'something_went_wrong';
      }
    } catch (e) {
      permitServerError = 'something_went_wrong';
    }
  }

  props.actions.permitServerError.update(permitServerError);
  props.actions.permitSubmitting.update(false);
  if (permitServerError) {
    scrollToElement('error-message-div', 'id');
  }
};

const getPermitFormChangeHandler = (state: any) => (
  props: IVariables,
  serviceName: string,
  fieldValues: IVariables,
) => {
  const formValues = state.permitInfo[serviceName];
  props.actions.permitInfo.update({
    ...state.permitInfo,
    [serviceName]: {
      ...formValues,
      ...fieldValues,
    },
  });
};

const handlePermitFormChange = (state: any) => (
  props: IVariables,
  value: any,
) => {
  const { permitInfo } = props;
  const { serviceType } = state;
  props.actions.permitInfo.update({
    ...permitInfo,
    [serviceType]: {
      ...permitInfo[serviceType],
      ...value,
    },
  });
};

const handleCategoryChange = (props: IVariables, cat: string) => {
  props.actions.permitType.update(cat);
  props.actions.serviceType.update('');
  props.actions.companyType.update('DED');
};

const handleServiceTypeChange = (props: IVariables, permit: string) => {
  props.actions.serviceType.update(permit);
  props.actions.companyType.update('DED');
  const { permitInfo } = props;
  if (props.user) {
    const Info = {
      name: props.user['First Name EN'],
      email: props.user['User Email'],
      phone: props.user.Mobile,
    };

    props.actions.permitInfo.update({
      ...permitInfo,
      [permit]: {
        ...permitInfo[permit],
        applicantContact: {
          ...Info,
        },
      },
    });
  }
};

const handleCompanyTypeChange = (props: IVariables, companyId: string) => {
  props.actions.companyType.update(companyId);
  props.actions.companyDetails.reset();
};

const handleCompanyDetailsChange = (props: IVariables, value: any) => {
  const { companyDetails } = props;
  props.actions.companyDetails.update({
    ...companyDetails,
    ...value,
  });
};

const handleRepresentativeTypeChange = (props: IVariables, value: any) => {
  const { companyDetails } = props;
  const { representativeType } = value;
  if (representativeType === '1') {
    props.actions.companyDetails.update({
      ...companyDetails,
      ...value,
      partnerSharePercentage: '100',
    });
  } else if (representativeType === '2') {
    props.actions.companyDetails.update({
      ...companyDetails,
      ...value,
      partnerSharePercentage: '1',
    });
  } else {
    props.actions.companyDetails.update({
      ...companyDetails,
      ...value,
      partnerSharePercentage: '0',
    });
  }
};

const handleValidityPeriodChange = (props: IVariables, value: any) => {
  const { validityPeriod } = props;
  props.actions.validityPeriod.update({
    ...validityPeriod,
    ...value,
  });
};
const getFormFields = (state: any) => () => {
  const { serviceType } = state;
  const PermitConfigs: { [key: string]: any } = permitConfigs;
  const permitConfig = PermitConfigs[serviceType];

  const { formFields }: { formFields: IVariables } = permitConfig;

  return formFields;
};

export default {
  init,
  onSubmit,
  handlePermitFormChange,
  handleCategoryChange,
  handleCompanyTypeChange,
  handleCompanyDetailsChange,
  handleRepresentativeTypeChange,
  handleServiceTypeChange,
  getPermitFormChangeHandler,
  getFormFields,
  validation,
  fetchLicenseDetails,
  handleValidityPeriodChange,
};
