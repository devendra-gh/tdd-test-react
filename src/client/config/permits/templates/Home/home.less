@import '~@tamm/ui-lib-v2-styles/colors.less';

.ui-lib-related-journey-card__content {
  svg {
    margin-bottom: 1rem;

    path {
      fill: white;
    }
  }
}

.PermitForm {
  &__form {
    &-tac-ui {
      padding: 0;
      margin-bottom: '3rem';
    }
  }

  &__button {
    &-group {
      display: flex;
      align-items: center;

      &-mobile {
        margin-top: 2rem;
        display: flex;
        flex-direction: column;
      }
    }

    &-cancel {
      text-transform: capitalize;
      font-weight: 500;
      text-decoration: underline;
      color: @ui-lib-link-color;
      margin-left: 2rem;
      cursor: pointer;

      &-mobile {
        margin-left: 0;
        margin-top: 2rem;
      }
    }
  }
}

.ui-lib-locale-provider_rtl .PermitForm__button-cancel {
  margin-right: 2rem;
}

ul.PermitForm__form-tac-ui {
  margin-bottom: 0;
  color: @ui-lib-body-text-color;
  line-height: 1.6;
  font-size: 1.5;
  padding-left: 0 !important;
}

.ui-lib-viewport ul.PermitForm__form-tac-ui li {
  display: flex;
  flex-direction: row;
  font-size: 16px;
  list-style-type: none;
}

.ui-lib-viewport ul.PermitForm__form-tac-ui li::before {
  // font-size: 1.5rem;
  padding: 0 1em;
  padding-top: 1.3rem;
  top: 0;
}

.ui-lib-link {
  margin-left: 2rem;
}
