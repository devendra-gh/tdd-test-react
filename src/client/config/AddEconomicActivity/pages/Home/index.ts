import { IVariables } from '@tamm/app-composer';
import { PATH_HOME } from 'client/config/AddEconomicActivity/routes';
import functions from './functions';

const home = [
  {
    path: PATH_HOME, // path for router
    uniqueId: 'add-economic-activity.home', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'addEconomicActivity.title', // title of the page, later it will be read from CMS
    init: functions.init,
    props: {
      title: 'addEconomicActivity.home.title',
      description: 'addEconomicActivity.home.description',
      showRelatedJourneyCard: true,
      startLogin: {
        title: 'addEconomicActivity.title',
        description: 'addEconomicActivity.home.startLogin.description',
        onClick: functions.onClick,
        buttonLabel: 'button.start',
      },
      process: {
        title: 'addEconomicActivity.home.process.title',
        steps: [
          {
            label: 'addEconomicActivity.addActivity',
            description: 'addEconomicActivity.addActivity.description',
          },
          {
            label: 'addEconomicActivity.getDEDApproval',
            description: 'addEconomicActivity.getDEDApproval.description',
          },
        ],
      },
      helpfulBlock: {
        callField: {
          onChange: () => {},
        },
        commentField: {
          onChange: () => {},
        },
        emailField: {
          onChange: () => {},
        },
        onChange: () => {},
        submitButton: {
          onClick: () => {},
        },
        telephoneField: {
          countries: [],
          onChange: () => {},
        },
      },
    },
    state: {
      mapState: [
        'stepsStatus',
        {
          steps: (state: IVariables) => {
            return [];
          },
        },
      ],
      mapDispatch: [
        'stepsStatus',
        'formData',
        'newActivityApiData',
        'helperData',
        'instanceId',
        'businessKey',
      ],
    },
  },
];

export default home;
