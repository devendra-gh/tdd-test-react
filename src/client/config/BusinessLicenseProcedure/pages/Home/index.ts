// import { REQUIRES_LOGIN } from '@tamm/app-composer';
import functions from './functions';

const home = [
  {
    path: ['/business-licence-procedure/'], // path for router
    uniqueId: 'licence-submit', // uniqueId for caching and other purposes
    template: 'businessLicenseProcedureHome', // template name, must be located in index of folder template/index
    title: 'businessLicenseProcedure.main.title', // title of the page, later it will be read from CMS
    props: {
      onStart: functions.onStart,
    },
    state: {
      mapState: [
        'loggedIn',
        'user',
        'instanceId',
        'businessKey',
        {
          steps: () => [],
        },
      ],
      mapDispatch: [
        'instanceId',
        'businessKey',
        'businessLicenseProcedureSteps',
      ],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default home;
