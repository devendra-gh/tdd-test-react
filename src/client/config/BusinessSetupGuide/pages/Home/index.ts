import functions from './functions';

const home = [
  {
    path: ['/investment-compass/'], // path for router
    uniqueId: 'home-page', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'bsg.home.title', // title of the page, later it will be read from CMS
    init: functions.init,
    onPageInit: functions.onPageInit,
    translations: {
      dictionaryId: 'buy-new-car',
      templateId: '4f3b7eee7185484f988ae2d898d666ba',
    },
    props: {
      breadcrumbs: [
        {
          label: 'bsg.breadcrumb.level0.title',
          link: 'https://www.tamm.abudhabi/',
        },
        {
          label: 'bsg.breadcrumb.level1.title',
          link: 'https://www.tamm.abudhabi/journeys/start-your-business',
        },
      ],
    },
    state: {
      mapState: [],
    },
  },
];

export default home;
