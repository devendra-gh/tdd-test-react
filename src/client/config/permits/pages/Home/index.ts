import {
  PROCESS_NAME,
  REQUIRES_SOP3,
  REQUIRES_CUSTOM_LOGIN,
} from 'client/config/permits/constants';
import functions from './functions';
import { PATH_APPLICATION_DETAILS } from '../../utils/constants/pageRoutes';

const enterPermitDetails = [
  {
    path: `/:serviceName${PATH_APPLICATION_DETAILS}`,
    uniqueId: 'application-submit',
    template: 'permitForm',
    title: 'pageTitle.permitApplication',
    init: functions.init,

    props: {
      onSubmit: functions.onSubmit(PROCESS_NAME),
      onSubmitLabel: 'Submit',
      handleCategoryChange: functions.handleCategoryChange,
      handleServiceTypeChange: functions.handleServiceTypeChange,
      handleCompanyTypeChange: functions.handleCompanyTypeChange,
      handleCompanyDetailsChange: functions.handleCompanyDetailsChange,
      handleValidityPeriodChange: functions.handleValidityPeriodChange,
      handleRepresentativeTypeChange: functions.handleRepresentativeTypeChange,
      fetchLicenseDetails: functions.fetchLicenseDetails,
      validation: functions.validation,
      showUpload: true,
    },
    fromProcessState: {
      processName: PROCESS_NAME,
      variables: ['permitErrorMessage'],
    },
    state: {
      mapState: [
        'user',
        'instanceId',
        'businessKey',
        'loggedIn',
        'companyType',
        'permitCompanyTypes',
        'companyDetails',
        'permitType',
        'serviceType',
        'pageTitle',
        'permitInfo',
        'permitSubmitting',
        'permitServerError',
        'urlServiceName',
        {
          handlePermitFormChange: functions.handlePermitFormChange,
          getFormFields: functions.getFormFields,
        },
      ],
      mapDispatch: [
        'instanceId',
        'businessKey',
        'serviceType',
        'pageTitle',
        'permitInfo',
        'permitType',
        'permitCompanyTypes',
        'companyType',
        'companyDetails',
        'permitSubmitting',
        'permitServerError',
        'urlServiceName',
      ],
    },
    requires: [REQUIRES_CUSTOM_LOGIN, REQUIRES_SOP3],
  },
];

export default enterPermitDetails;
