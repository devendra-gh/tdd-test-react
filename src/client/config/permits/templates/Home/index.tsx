import React from 'react';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import ServiceTemplate from '@tamm/ui-lib-v2-service-template';
import Sidebar from 'client/templates/PermitsSidebar';
import permitConfigs from 'client/config/permits/permitConfigs';
import './home.less';

/* istanbul ignore file */
/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const {
    i18n,
    // title,
    serviceType,
  } = props;
  const PermitConfigs: { [key: string]: any } = permitConfigs;
  const serviceConfig = PermitConfigs[serviceType];
  let serviceDescription = {
    conditions: [],
    title: '',
  };
  let documents: string[] = [];
  let documentsList: { description: any; id: string }[] = [];
  let fees: string[] = [];
  let feesList: { fees: any; id: string }[] = [];

  if (serviceConfig) {
    serviceDescription = serviceConfig.landingPage.serviceDescription;
    documents = serviceConfig.landingPage.documents;
    fees = serviceConfig.landingPage.fees;
    if (Array.isArray(documents)) {
      documentsList = documents.map((document: string, key: number) => {
        return { description: i18n(document), id: key.toString() };
      });
    } else {
      const documentCollections: Record<string, IVariables> = documents;
      Object.values(documentCollections).forEach(
        (documentCollection: IVariables, key: number) => {
          documentsList.push({
            description: i18n(documentCollection.title),
            id: key.toString(),
          });
          documentCollection.collection.forEach(
            (document: IVariables, index: number) => {
              documentsList.push({
                description: `${(index + 1).toString()}.  ${i18n(
                  document.label,
                )}`,
                id: `${key.toString()}_${index.toString()}`,
              });
            },
          );
          return documentsList;
        },
      );
    }

    feesList = fees.map((fee: string, key: number) => {
      return { fees: i18n(fee), id: key.toString() };
    });
  }

  const items = [
    {
      description: i18n('process.submitApplicationDesc'),
      label: i18n('process.submitApplication'),
    },
    {
      description: i18n('process.dedApprovalDesc'),
      label: i18n('process.dedApproval'),
    },
    {
      description: i18n('process.paymentDesc'),
      label: i18n('process.payment'),
    },
    {
      description: i18n('process.downloadPermitDesc'),
      label: i18n('process.downloadPermit'),
    },
  ];
  const newitems = [
    {
      description: i18n('process.submitApplicationDesc'),
      label: i18n('process.submitApplication'),
    },
    {
      description: i18n('process.dedCheckDesc'),
      label: i18n('process.dedCheck'),
    },
    {
      description: i18n('process.entityApprovalDesc'),
      label: i18n('process.entityApproval'),
    },
    {
      description: i18n('process.getDedApprovalDesc'),
      label: i18n('process.getDedApproval'),
    },
  ];

  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <Sidebar
            currentStep={props.currentStep}
            currentSubStep={props.currentSubStep}
            i18n={props.i18n}
            steps={[]}
            stepsStatus={props.stepsStatus}
            hideSidebar={props.hideSidebar}
          />
        }
      >
        {
          <ServiceTemplate
            description={
              <div>
                <span>{i18n(serviceDescription.title)}</span>
                <br />
                <br />
                <ul className="PermitForm__form-tac-ui">
                  {serviceDescription.conditions.map(
                    (condition: string, key: number) => (
                      <li key={String(key)}>{i18n(condition)}</li>
                    ),
                  )}
                </ul>
              </div>
            }
            helpfulBlock={{
              onFieldChange: function noRefCheck() {},
              onSubmit: function noRefCheck() {},
              telephoneField: {
                countries: [
                  {
                    code: 971,
                    name: 'UAE',
                  },
                ],
              },
            }}
            process={{
              steps:
                serviceType === 'mobile-car-tajer-permit-in-abu-dhabi' ||
                serviceType === 'food-truck-permit-in-abu-dhabi'
                  ? newitems
                  : items,
              title: i18n('process.title'),
            }}
            startLogin={{
              buttonLabel: i18n('landingPage.buttonStart'),
              // description: i18n(
              //   `${
              //     serviceType
              //       ? `startLogin.permitApplication.${serviceType}`
              //       : `${title}`
              //   }`,
              // ),
              title: i18n('landingPage.mainTitle'),
              onClick: () => props.onStart(props),
            }}
            tables={[
              {
                columns: [
                  {
                    id: 'description',
                    title: i18n('processTable.document.tableHeader'),
                  },
                ],
                items: documentsList,

                title: i18n('processTable.document.title'),
              },
              {
                columns: [
                  {
                    id: 'fees',
                    title: i18n('processTable.fees.tableHeader'),
                  },
                ],
                items: feesList,
                title: i18n('processTable.fees.title'),
              },
            ]}
            title={i18n(props.serviceTitle)}
          />
        }
        <div style={{ height: 60 }} />
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
