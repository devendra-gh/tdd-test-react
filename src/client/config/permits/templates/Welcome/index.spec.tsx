import React, { FunctionComponent } from 'react';
import { shallow } from 'enzyme';
import Home from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: FunctionComponent) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('config/templates/Welcome', () => {
  let props: any;

  beforeEach(() => {
    props = {
      match: {},
      location: {},
      locale: {
        switch: jest.fn(),
      },
      i18n: jest.fn(),
      title: 'title',
      buttons: [
        {
          label: 'test_button',
          uiType: 'primary',
          withArrow: true,
          onClick: jest.fn(),
        },
      ],
    };
  });

  it('should set welcome page', async () => {
    const component = shallow(<Home {...props} />);
    // const buttonClick = component.find('Button');
    // buttonClick.simulate()
    expect(component).toMatchSnapshot();
  });
});
