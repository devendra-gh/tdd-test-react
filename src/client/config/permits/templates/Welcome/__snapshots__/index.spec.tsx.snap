// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/templates/Welcome should set welcome page 1`] = `
<Container>
  <div
    style={
      Object {
        "marginBottom": "8rem",
      }
    }
  >
    <withAnalytics(LocalizedComponent)
      analyticsConfig={
        Object {
          "eventsFrom": Array [],
        }
      }
      aria-label="test_button"
      onClick={[Function]}
      withArrow={true}
    />
    <br />
    <br />
  </div>
</Container>
`;
